Fixes for troubleshooting guide re. Hubble/Hubble Relay (#13644)
cilium-team/cilium: Documentation/operations/troubleshooting.rst

       Allows from labels {"matchLabels":{"any:org":"empire","k8s:io.kubernetes.pod.namespace":"default"}}
         No label match for [k8s:class=xwing k8s:io.cilium.k8s.policy.cluster=default k8s:io.cilium.k8s.policy.serviceaccount=default k8s:io.kubernetes.pod.namespace=default k8s:org=alliance]
   1/1 rules selected
   Found no allow rule
   Ingress verdict: denied

   Final verdict: DENIED

Please refer to the :ref:`policy troubleshooting guide<policy_tracing>` for
more details about how to troubleshoot policy related drops.

.. note::
   **Hubble Relay**  allows you to query multiple Hubble instances
   simultaneously without having to first manually target a specific node.  See
   `Observing flows with Hubble Relay`_ for more information.

Ensure Hubble is running correctly
----------------------------------

To ensure the Hubble client can connect to the Hubble server running inside
Cilium, you may use the ``hubble status`` command:

.. code:: shell-session

   $ hubble status
   Healthcheck (via unix:///var/run/cilium/hubble.sock): Ok
   Max Flows: 4096
   Current Flows: 2542 (62.06%)

> ``cilium-agent`` must be running with the ``--enable-hubble`` option in order

Reviewer: ```suggestion
``cilium-agent`` must be running with the ``--enable-hubble`` option (default) in order
```

To check if Hubble is enabled in your deployment, you may look for the
following output in ``cilium status``:

.. code:: shell-session

   $ cilium status
   ...
   Hubble:   Ok   Current/Max Flows: 2542/4096 (62.06%), Flows/s: 164.21   Metrics: Disabled
   ...

.. note::
   Pods need to be managed by Cilium in order to be observable by Hubble.
   See how to :ref:`ensure a pod is managed by Cilium<ensure_managed_pod>`
   for more details.

Observing flows with Hubble Relay
=================================

Hubble Relay is a service which allows to query multiple Hubble instances
simultaneously and aggregate the results. As Hubble Relay relies on individual

> Hubble instances, Hubble needs to be enabled when deploying Cilium. In

Reviewer: As Hubble is enabled by default in 1.9, this can be simplified a bit:

> Hubble Relay is a service which allows to query multiple Hubble instances simultaneously and aggregate the results. The Hubble service needs to be exposed on TCP port 4244 to allow Hubble Relay to connect to individual instances. This can be done via Helm values  or option ``--hubble-listen-address :4244`` on cilium-agent.

Also, the note below about security implications can be removed as the endpoint is now secured via mTLS by default.

.. code:: bash

   hubble status --server localhost:4245

This command should return an output similar to the following:

.. code:: shell-session

   Healthcheck (via localhost:4245): Ok
   Max Flows: 16384
   Current Flows: 16384 (100.00%)

> For convenience, you may set and export the ``HUBBLE_DEFAULT_SOCKET_PATH``

Reviewer: ```suggestion
For convenience, you may set and export the ``HUBBLE_SERVER``
```

The old environment variable (`HUBBLE_DEFAULT_SOCKET_PATH`) still works but is deprecated/not recommended anymore.

environment variable:

.. code:: bash

>    export HUBBLE_DEFAULT_SOCKET_PATH=localhost:4245

Reviewer: ```suggestion
   export HUBBLE_SERVER=localhost:4245
```
